apps/meteor/app/livechat/server/lib/RoutingManager.js

```diff
@@ -196,7 +196,8 @@ export const RoutingManager = {
 
 		if (!agent) {
 			logger.debug(`Cannot take Inquiry ${inquiry._id}: Precondition failed for agent`);
-			return callbacks.run('livechat.onAgentAssignmentFailed', { inquiry, room, options });
+			await callbacks.run('livechat.onAgentAssignmentFailed', { inquiry, room, options });
```

Comment:
The value from the callback won't get passed to `forwardRoomToDepartment` in this case since you're not returning it again in this method. Perhaps you can change it to:

```
const value = await callbacks.run('livechat.onAgentAssignmentFailed', { inquiry, room, options });
return value;
```